client/main.go

```diff
@@ -465,6 +465,77 @@ func schemeUnderstood(scheme string) error {
 	return nil
 }
 
+// TODO function header
+func DoList(ctx context.Context, remoteObject string, flagOptions map[string]bool, options ...TransferOption) (err error) { //TODO change from TransferOption to something with ls
```

Comment:
Don't use two separate option mechanisms -- use the `TransferOption` approach from elsewhere in the library.